### PR TITLE
Added 'regolith tool' command

### PR DESCRIPTION
**Pull request by Nusiq**
This PR adds `regolith tool` command.

Syntax:
```
regolith tool <filter-name> [arguments...]

filter-name - the name of the filter to run
argumetns - the arguments passed to the filter
```
Note that the filters that use the "settings" property are also supported, you can simply pass the
JSON for settings as a first argument. Internally there is no difference between using
the first argument of "arguments" or using "settings". Settings is just more convenient when
used through the config.json file.

The `regolith tool` command, runs a single filter and exports created files into source RP, BP and data. **This operation is destructive.** If filter fails, the files are not exported. If something goes wrong during exporting, the `RevertibleFsOperations` should be able to Undo the export operation. This means that the files will be modified only if everything will run without errors.

### CODE REVIEW

**Comment by Nusiq** [comment]
Note that the `regolith help` command for `regolith tool` is not a part of this commit. I want to write that as a part of PR #219 (which improves CLI in general). After merging this I will probably merge `develop` branch to the branch that improves CLI.

File: docs/_pages/docs/getting_started/getting-started.md [hunk 1]
```diff
@@ -89,12 +89,18 @@ The `regolith watch [profile-name]` command works the same as `regolith run`, bu
 it will watch your source files and rerun the profile when they change. If you're
 using `regolith run` you have to do it manually every time.
 
-A single run copies your source files into a temporary folder, runs all of the
-filters of the profile and moves the files to target location defined in the
-"export" property of the profile. By default the export is set to "development",
-which means that the files will be copied to the `development` pack folders of
-`com.mojang`. The names of folders created in this export mode are based on
-the name of the project like `project_name_bp` and `project_name_rp`.
+A single run copies your source files into a temporary folder, then runs all
```
Comment (SirLich) [inline docs/_pages/docs/getting_started/getting-started.md:92]: This paragraph could be rewritten to be a list, like 1) 2) 3) (bullet points) to make it easier to see the order.

**Review by SirLich — COMMENTED** [review]
Alongside the two comments I left in my review, I would additionally suggest making a *new page* dedicated to the new tool command. Do this wherever 'regolith run/watch' is documented. 

It would also be good to make a small note/warning in the 'developing a filter' section where you explain that filters can be run destructively. 

If this PR contains handling for filter.json to define a 'tool' section (?), then naturally this also needs to be documented.

File: docs/_pages/docs/getting_started/getting-started.md (hunk 1 above)
Comment (Nusiq) [reply docs/_pages/docs/getting_started/getting-started.md]: Done. Except that this text was moved to the `filter-run-modes.md`.

File: docs/_pages/docs/getting_started/getting-started.md [hunk 2]
```diff
@@ -139,6 +145,22 @@ Check `com.mojang`, and open the new `texture_list.json` file in `RP/textures/te
 {: .notice--warning}
 `Warning:` If your project doesn't have any textures, than `texture_list.json` will simply create a blank file `[]`. Consider adding some textures to see the filter at work!
 
+## Running Regolith Destructively
```
[the remark this answers is not in the document]
Comment (Nusiq) [reply docs/_pages/docs/getting_started/getting-started.md]: This section was moved to a new file called `filter-run-modes.md`

**Review by stirante — APPROVED** [review]

**Pull request merged**